src/Compilers/CSharp/Portable/BoundTree/BoundDecisionDagNode.cs

```diff
@@ -46,5 +52,73 @@ public override int GetHashCode()
                     throw ExceptionUtilities.UnexpectedValue(this);
             }
         }
+
+#if DEBUG
+        private int _id = -1;
+
+        public int Id
+        {
+            get
+            {
+                return _id;
+            }
+            internal set
+            {
+                Debug.Assert(value >= 0, "Id must be non-negative but was set to " + value);
+                Debug.Assert(_id == -1, $"Id was set to {_id} and set again to {value}");
+                _id = value;
+            }
+        }
+
+        internal new string GetDebuggerDisplay()
+        {
+            var pooledBuilder = PooledStringBuilder.GetInstance();
```

Comment:
>var pooledBuilder = PooledStringBuilder.GetInstance(); [](http://example.com/codeflow?start=12&length=54)

Perhaps just create a `new StringBuilder()` instead, to avoid the extra complexity evaluating in the debugger.